src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/singlespaceseparator/InputSingleSpaceSeparatorWithEmoji.java

```diff
@@ -0,0 +1,45 @@
+/*
+SingleSpaceSeparator
+validateComments = (default)false
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.whitespace.singlespaceseparator;
+
+public class InputSingleSpaceSeparatorWithEmoji {
+
+    String s1 = "🧐😉           assa  "; // ok
+    String s2 = "🧐😉" + "          " + "assa  "; // ok
+    String s3 =  "🧐" + "🎄 "; // violation 'Use a single space to separate non-whitespace characters'
+    String s4
+            =   "da🎄"; // violation 'Use a single space to separate non-whitespace characters'
+    private void foo(String s) {
+      if (s.equals("🤩🎄")  ){ // violation 'Use a single space to separate non-whitespace characters'
+                foo(s);
+      }
+    }
+
+    /* always correct */ /*🧐*/ String s = "🎄😉";
+    String /* 🧐  wrong if X is enabled */   st =  "🎄  assa";
+    // violation above 'Use a single space to separate non-whitespace characters'
+    String str = "🤩🎄";   // Multiple whitespaces before comment, Wrong if X is enabled
```

Comment:
`Wrong if X is enabled` what does this mean? Does this refer to `validateComments` property?

Reply:
Yes

Reply:
This is confusing, we should stick to `ok` and `violation, ...` comments only, unless example is very complicated and requires additional explanation.

Reply:
Resolved.